src/TSID/src/QPFixedBaseTSID.cpp

```diff
@@ -0,0 +1,601 @@
+    }
+
+    return tasksName;
+}
+
+std::weak_ptr<System::LinearTask> QPFixedBaseTSID::getTask(const std::string& name) const
+{
+    constexpr auto logPrefix = "[QPFixedBaseTSID::getTask]";
+    auto task = m_pimpl->tasks.find(name);
+
+    if (task == m_pimpl->tasks.end())
+    {
+        log()->warn("{} The task named {} does not exist. A nullptr will be returned.",
+                    logPrefix,
+                    name);
+        return std::shared_ptr<System::LinearTask>(nullptr);
+    }
+
+    return task->second.task;
+}
+
+bool QPFixedBaseTSID::setKinDyn(std::shared_ptr<iDynTree::KinDynComputations> kinDyn)
+{
+    constexpr auto logPrefix = "[QPFixedBaseTSID::setKinDyn]";
+
+    if (kinDyn == nullptr)
+    {
+        log()->error("{} The kinDyn object is not valid.", logPrefix);
+        return false;
+    }
+
+    if (!m_pimpl->baseSE3Task->setKinDyn(kinDyn))
+    {
+        log()->error("{} Unable to set the kinDyn object for SE3Task on the base.", logPrefix);
+        return false;
+    }
+
+    if (!m_pimpl->dynamicsTask->setKinDyn(kinDyn))
+    {
+        log()->error("{} Unable to set the kinDyn object for dynamicsTask.", logPrefix);
+        return false;
+    }
+
+    m_pimpl->baseLink = kinDyn->getFloatingBase();
+
+    m_pimpl->isKinDynSet = true;
+
+    return true;
+}
+
+bool QPFixedBaseTSID::createBaseSE3Task()
+{
+    constexpr auto logPrefix = "[QPFixedBaseTSID::createBaseSE3Task]";
+
+    auto baseSE3ParameterHandler = std::make_shared<StdImplementation>();
+    baseSE3ParameterHandler->setParameter("robot_acceleration_variable_name",
+                                          m_pimpl->robotAccelerationVariable.name);
+    baseSE3ParameterHandler->setParameter("kp_linear", 0.0);
+    baseSE3ParameterHandler->setParameter("kd_linear", 0.0);
+    baseSE3ParameterHandler->setParameter("kp_angular", 0.0);
+    baseSE3ParameterHandler->setParameter("kd_angular", 0.0);
+    baseSE3ParameterHandler->setParameter("frame_name", m_pimpl->baseLink);
+
+    if(!m_pimpl->baseSE3Task->initialize(baseSE3ParameterHandler))
+    {
+        log()->error("{} Error initializing se3task on the base", logPrefix);
+        return false;
+    }
+
+    m_pimpl->baseSE3Task->setSetPoint(manif::SE3d::Identity(),
+                                      manif::SE3d::Tangent::Zero(),
+                                      manif::SE3d::Tangent::Zero());
+
+    return true;
+}
+
+bool QPFixedBaseTSID::createDynamicsTask()
+{
+    constexpr auto logPrefix = "[QPFixedBaseTSID::createDynamicsTask]";
+
+    auto dynamicsParameterHandler = std::make_shared<StdImplementation>();
+
+    dynamicsParameterHandler->setParameter("robot_acceleration_variable_name",
+                                           m_pimpl->robotAccelerationVariable.name);
+    dynamicsParameterHandler->setParameter("joint_torques_variable_name", m_pimpl->jointTorquesVariable.name);
+    dynamicsParameterHandler->setParameter("max_number_of_contacts", 1);
```

Comment:
Could you add documentation as to why we hard-code this to 1? Also in the header file you can reference to this setting saying that, since its FixedBaseQP, we assume the external wrench acting on only the base link.